lib/galaxy/objectstore/s3_multipart_upload.py

```diff
@@ -14,17 +14,8 @@
 
 from multiprocessing.pool import IMapIterator
 
-try:
-    from galaxy import eggs
-    eggs.require('boto')
-except ImportError:
-    pass
-
-try:
-    import boto
-    from boto.s3.connection import S3Connection
-except ImportError:
-    boto = None
+import boto
+from boto.s3.connection import S3Connection
```

Comment:
Switching it back (in fact, `galaxy.objectstore.s3` already handles this correctly, but it probably would not have worked if you actually loaded the S3 object store, since `.s3_multipart_upload` was imported before `boto`). It doesn't look like we should need a `if boto is None` handler in here since that will already be done by the S3 object store.